src/openrct2-ui/windows/RideConstruction.cpp

```diff
@@ -3616,13 +3616,12 @@ void ride_construction_toolupdate_construct(ScreenCoordsXY screenCoords)
         && ride_type_has_flag(ride->type, RIDE_TYPE_FLAG_IS_SHOP))
     {
         TileElement* pathsByDir[4];
-        constexpr TileCoordsXY DirOffsets[4] = { { -1, 0 }, { 0, 1 }, { 1, 0 }, { 0, -1 } };
 
         bool keepOrientation = false;
         for (int8_t i = 0; i < 4; i++)
         {
-            pathsByDir[i] = map_get_footpath_element(
-                (mapCoords->x >> 5) + DirOffsets[i].x, (mapCoords->y >> 5) + DirOffsets[i].y, z >> 3);
+            pathsByDir[i] = map_get_footpath_element({ mapCoords->x + (DirectionOffsets[i].x * COORDS_XY_STEP),
```

Comment:
Use the CoordsXY direction delta instead then you don't need to multiply and can use the + overload

Reply:
Nice, I wasn't too happy about these lines either. :+1:

Reply:
`CoordsDirectionDelta` is the variable you should use. `pathsByDir[i] = map_get_footpath_element({ *mapCoords + CoordsDirectionDelta[i], z });